main.go

```diff
@@ -173,14 +153,105 @@ func wrapMain() error {
 
 }
 
+func configFromStdin(vargs *GAE, workspace *string) error {
+	// https://godoc.org/github.com/drone/drone-plugin-go/plugin
+	workspaceInfo := plugin.Workspace{}
+	plugin.Param("workspace", &workspaceInfo)
+	plugin.Param("vargs", vargs)
+	// Note this hangs if no cli args or input on STDIN
+	plugin.MustParse()
+
+	*workspace = workspaceInfo.Path
+
+	return nil
+}
+
+// GAE struct has different json for these, so use an intermediate for the new drone format
+type dummyGAE struct {
+	AddlArgs map[string]string `json:"-"`
+	AEEnv    map[string]string `json:"-"`
+}
+
+func configFromEnv(vargs *GAE, workspace *string) error {
+
+	// drone plugin input format du jour:
+	// http://readme.drone.io/plugins/plugin-parameters/
+
+	// Strings
+	vargs.Project = os.Getenv("PLUGIN_PROJECT")
+	vargs.Action = os.Getenv("PLUGIN_ACTION")
+	*workspace = os.Getenv("DRONE_WORKSPACE")
+	vargs.Token = os.Getenv("GAE_CREDENTIALS") // secrets are not prefixed
+	vargs.Version = os.Getenv("PLUGIN_VERSION")
+	vargs.FlexImage = os.Getenv("PLUGIN_FLEX_IMAGE")
+	vargs.AppFile = os.Getenv("PLUGIN_APP_FILE")
+	vargs.CronFile = os.Getenv("PLUGIN_CRON_FILE")
+	vargs.DispatchFile = os.Getenv("PLUGIN_DISPATCH_FILE")
+	vargs.Dir = os.Getenv("PLUGIN_DIR")
+	vargs.AppCfgCmd = os.Getenv("PLUGIN_APPCFG_CMD")
+	vargs.GCloudCmd = os.Getenv("PLUGIN_GCLOUD_CMD")
```

Comment:
Next time, you may save some finger work with this lib: https://github.com/kelseyhightower/envconfig